JaegerRemoteSamplerTest is flaky. (Fail 5x)

see https://circleci.com/gh/open-telemetry/opentelemetry-java/4241 or
https://circleci.com/gh/open-telemetry/opentelemetry-java/4259 for example

```
> Task :opentelemetry-sdk-contrib-jaeger-remote-sampler:test
Picked up _JAVA_OPTIONS: -Xmx3g

io.opentelemetry.sdk.contrib.trace.jaeger.sampler.JaegerRemoteSamplerTest > description FAILED
    java.lang.AssertionError: Resource ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=9, target=directaddress:///my-service}} can not be released in time at the end of test
        at io.grpc.testing.GrpcCleanupRule.teardown(GrpcCleanupRule.java:169)
        at io.grpc.testing.GrpcCleanupRule.access$100(GrpcCleanupRule.java:48)
        at io.grpc.testing.GrpcCleanupRule$1.evaluate(GrpcCleanupRule.java:138)
```

Re-running the build usually helps but this should be fixed.


Tracking instances:
https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-java/2325/workflows/8072b109-a9cd-4313-adcc-7304d4d51d49/jobs/4731/steps

https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-java/2335/workflows/c823a3cc-0c48-45db-ae87-646be91f033a/jobs/4754

--- a/sdk_contrib/jaeger_remote_sampler/src/test/java/io/opentelemetry/sdk/contrib/trace/jaeger/sampler/JaegerRemoteSamplerTest.java
+++ b/sdk_contrib/jaeger_remote_sampler/src/test/java/io/opentelemetry/sdk/contrib/trace/jaeger/sampler/JaegerRemoteSamplerTest.java
@@ -38,6 +38,7 @@ import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 import org.awaitility.Awaitility;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -50,6 +51,10 @@ public class JaegerRemoteSamplerTest {
 
   private static final String SERVICE_NAME = "my-service";
   private static final int RATE = 999;
+
+  private final String serverName = InProcessServerBuilder.generateName();
+  private final ManagedChannel inProcessChannel =
+      InProcessChannelBuilder.forName(serverName).directExecutor().build();
 
   @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
 
@@ -79,24 +84,27 @@ public class JaegerRemoteSamplerTest {
     }
   }
 
-  @Test
-  public void connectionWorks() throws IOException {
-    final String serverName = InProcessServerBuilder.generateName();
-    final ArgumentCaptor<SamplingStrategyParameters> requestCaptor =
-        ArgumentCaptor.forClass(Sampling.SamplingStrategyParameters.class);
-
+  @Before
+  public void before() throws IOException {
     grpcCleanup.register(
         InProcessServerBuilder.forName(serverName)
             .directExecutor()
             .addService(service)
             .build()
             .start());
+    grpcCleanup.register(inProcessChannel);
+  }
 
-    ManagedChannel channel =
-        grpcCleanup.register(InProcessChannelBuilder.forName(serverName).directExecutor().build());
+  @Test
+  public void connectionWorks() {
+    ArgumentCaptor<SamplingStrategyParameters> requestCaptor =
+        ArgumentCaptor.forClass(Sampling.SamplingStrategyParameters.class);
 
     JaegerRemoteSampler sampler =
-        JaegerRemoteSampler.newBuilder().setChannel(channel).setServiceName(SERVICE_NAME).build();
+        JaegerRemoteSampler.newBuilder()
+            .setChannel(inProcessChannel)
+            .setServiceName(SERVICE_NAME)
+            .build();
 
     Awaitility.await()
         .atMost(10, TimeUnit.SECONDS)
@@ -115,11 +123,11 @@ public class JaegerRemoteSamplerTest {
 
   @Test
   public void description() {
-    ManagedChannel channel =
-        grpcCleanup.register(
-            InProcessChannelBuilder.forName(SERVICE_NAME).directExecutor().build());
     JaegerRemoteSampler sampler =
-        JaegerRemoteSampler.newBuilder().setChannel(channel).setServiceName(SERVICE_NAME).build();
+        JaegerRemoteSampler.newBuilder()
+            .setChannel(inProcessChannel)
+            .setServiceName(SERVICE_NAME)
+            .build();
     assertTrue(
         sampler
             .getDescription()

